flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/fanout/FanOutProperties.java

```diff
@@ -0,0 +1,127 @@
+
+		efoRegistrationType = EFORegistrationType.valueOf(properties.getProperty(ConsumerConfigConstants.EFO_REGISTRATION_TYPE, EFORegistrationType.EAGER.toString()));
+		//if efo registration type is EAGER|LAZY, then user should explicitly provide a consumer name for each stream.
+		if (efoRegistrationType == EFORegistrationType.EAGER || efoRegistrationType == EFORegistrationType.LAZY) {
+			consumerName = properties.getProperty(ConsumerConfigConstants.EFO_CONSUMER_NAME);
+		} else {
+			//else users should explicitly provide consumer arns.
+			streamConsumerArns = new ArrayList<>();
+			for (String stream:streams) {
+				String key = ConsumerConfigConstants.EFO_CONSUMER_ARN_PREFIX + "." + stream;
+				streamConsumerArns.add(properties.getProperty(key));
+			}
+		}
+
+		this.subscribeToShardMaxRetries = Integer.parseInt(
+			properties.getProperty(
+				ConsumerConfigConstants.SUBSCRIBE_TO_SHARD_RETRIES,
+				Long.toString(ConsumerConfigConstants.DEFAULT_SUBSCRIBE_TO_SHARD_RETRIES)));
+		this.subscribeToShardBaseBackoffMillis = Long.parseLong(
+			properties.getProperty(ConsumerConfigConstants.SUBSCRIBE_TO_SHARD_BACKOFF_BASE,
+				Long.toString(ConsumerConfigConstants.DEFAULT_SUBSCRIBE_TO_SHARD_BACKOFF_BASE)));
+		this.subscribeToShardMaxBackoffMillis = Long.parseLong(
+			properties.getProperty(ConsumerConfigConstants.SUBSCRIBE_TO_SHARD_BACKOFF_MAX,
+				Long.toString(ConsumerConfigConstants.DEFAULT_SUBSCRIBE_TO_SHARD_BACKOFF_MAX)));
+		this.subscribeToShardExpConstant = Double.parseDouble(
+			properties.getProperty(ConsumerConfigConstants.SUBSCRIBE_TO_SHARD_BACKOFF_EXPONENTIAL_CONSTANT,
+				Double.toString(ConsumerConfigConstants.DEFAULT_SUBSCRIBE_TO_SHARD_BACKOFF_EXPONENTIAL_CONSTANT)));
+	}
+
+	public void setEfoRegistrationType(EFORegistrationType efoRegistrationType) {
+		this.efoRegistrationType = efoRegistrationType;
+	}
+
+	public void setConsumerName(@Nullable String consumerName) {
+		this.consumerName = consumerName;
+	}
+
+	public void setStreamConsumerArns(@Nullable List<String> streamConsumerArns) {
+		this.streamConsumerArns = streamConsumerArns;
+	}
+
+	public void setSubscribeToShardMaxRetries(int subscribeToShardMaxRetries) {
+		this.subscribeToShardMaxRetries = subscribeToShardMaxRetries;
+	}
+
+	public void setSubscribeToShardMaxBackoffMillis(long subscribeToShardMaxBackoffMillis) {
+		this.subscribeToShardMaxBackoffMillis = subscribeToShardMaxBackoffMillis;
+	}
+
+	public void setSubscribeToShardBaseBackoffMillis(long subscribeToShardBaseBackoffMillis) {
+		this.subscribeToShardBaseBackoffMillis = subscribeToShardBaseBackoffMillis;
+	}
+
+	public void setSubscribeToShardExpConstant(double subscribeToShardExpConstant) {
+		this.subscribeToShardExpConstant = subscribeToShardExpConstant;
+	}
+
+	public EFORegistrationType getEfoRegistrationType() {
+		return efoRegistrationType;
+	}
+
+	@Nullable
+	public String getConsumerName() {
+		return consumerName;
+	}
+
+	@Nullable
+	public List<String> getStreamConsumerArns() {
```

Comment:
@xiaolong-sn when we have a map we can add a method `public String getStreamConsumerArn(String stream)`

Reply:
Yes, I added one.